### PR TITLE
fix: can't drag scrollbar in window under treeland

### PR DESCRIPTION
**Pull request by FeiWang1119**
Add the logic about moving the window from platform plugin

Log: as title .
pms: BUG-284215

### CODE REVIEW

File: src/kernel/dplatformhandle.cpp [hunk 1]
```diff
@@ -635,39 +640,117 @@ class Q_DECL_HIDDEN CreatorWindowEventFilter : public QObject {
...
+        self->m_windowMoving = false;
+    }
+
+    if (is_mouse_move && !event->isAccepted()
+            && w->geometry().contains(static_cast<QMouseEvent*>(event)->globalPos())) {
+        if (!self->m_windowMoving && self->m_enableSystemMove) {
+            self->m_windowMoving = true;
+
+            event->accept();
+            static_cast<QPlatformWindow *>(w->handle())->startSystemMove();
```
[the remark this answers is not in the document]
Comment (FeiWang1119) [reply src/kernel/dplatformhandle.cpp]: done

File: src/kernel/dplatformhandle.cpp [hunk 2]
```diff
@@ -2,20 +2,28 @@
 //
 // SPDX-License-Identifier: LGPL-3.0-or-later
 
+#define protected public
```
Comment (kegechen) [inline src/kernel/dplatformhandle.cpp:5]: udpate license 2022 - 2024

**Comment by kegechen** [comment]
+1

File: src/kernel/dplatformhandle.cpp (hunk 2 above)
Comment (FeiWang1119) [reply src/kernel/dplatformhandle.cpp:5]: done

**Review by mhduiy — APPROVED** [review]

**Pull request merged**